include/dlaf/util_matrix.h

```diff
@@ -137,27 +137,27 @@ class getter_random<std::complex<T>> : private getter_random<T> {
 }
 
 /// Sets all the elements of all the tiles to zero
-template <class T, class ExecutorOrPolicy>
-void set0(ExecutorOrPolicy ex, Matrix<T, Device::CPU>& matrix) {
+template <class T, Device D, class ExecutorOrPolicy>
+void set0(ExecutorOrPolicy ex, Matrix<T, D>& matrix) {
   for (const auto& idx : iterate_range2d(matrix.distribution().localNrTiles()))
-    matrix(idx).then(ex, hpx::unwrapping(tile::internal::set0_o));
+    hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::internal::set0_o), matrix(idx));
 }
 
 /// Sets all the elements of all the tiles in the active range to zero
-template <class T, Coord axis, class ExecutorOrPolicy>
-void set0(ExecutorOrPolicy ex, Panel<axis, T, Device::CPU>& panel) {
+template <class T, Coord axis, Device D, class ExecutorOrPolicy>
+void set0(ExecutorOrPolicy ex, Panel<axis, T, D>& panel) {
   for (const auto& tile_idx : panel.iteratorLocal())
-    panel(tile_idx).then(ex, hpx::unwrapping(tile::internal::set0_o));
+    hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::internal::set0_o), panel(tile_idx));
```

Comment:
Very minor thing, just for future reference and awareness.

I had to switch from `.then` to `.dataflow`, because as @msimberg said

> the [cuda] executor only customizes for dataflow, not then, and we didn't consider it necessary to make it work with then back then, and even less so now with senders being closer to reality

> it has async_execute which makes hpx::async work, and it has our custom dataflow_finalize which makes hpx::dataflow work correctly, but not then_execute